src/context & data/UserContext.js

```diff
@@ -28,3 +28,43 @@ export const UserContextProvider = ({ children }) => {
 };
 
 export const useUserContext = () => useContext(UserContext);
+
+const GlobalReducer = (state, action) => {
+  switch (action.for) {
```

Comment:
I see you don't have any else cases (apart from default) for action.for. Then why not destructure for and just have the inner switch block? Is it possible to avoid nesting?

Reply:
The reason there is no other case is because those cases come under another pr review as they belong in different feature